### PR TITLE
Feature full pas template

### PR DESCRIPTION
**Pull request by mliora**

### CODE REVIEW

**Comment by avishayil** [comment]
Good job, please notice the following:
1. Please pay attention to word-wrapping, it seems we're using it a lot here and not all the times it provides better readability. Let's tune this to make the code look nicer.
2. Lambdas like the password generator and bucket copier is not needed anymore since we're handling this internally.

File: aws/Full-PAS-Deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,1607 @@
...
+    DeletionPolicy: Retain
+  DeployBucket:
+    Type: 'AWS::S3::Bucket'
+    Properties:
+      AccessControl: Private
+  CPMCfnInitLogStream:
+    Type: 'AWS::Logs::LogStream'
+    Properties:
+      LogGroupName: !Sub '${LogGroup}'
+      LogStreamName: CPMMachine/CPMCfnInitLogStream
```
[the remark this answers is not in the document]
Comment (yogevh) [reply aws/Full-PAS-Deployment.yaml]: if it doesn't mean that we need to create new AMIs

File: aws/Full-PAS-Deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,1607 @@
...
+    DeletionPolicy: Retain
+  CPMConfigurationsLog:
+    Type: 'AWS::Logs::LogStream'
+    Properties:
+      LogGroupName: !Sub '${LogGroup}'
+      LogStreamName: CPMMachine/CPMConfigurations
```
[the remark this answers is not in the document]
Comment (yogevh) [reply aws/Full-PAS-Deployment.yaml]: if it doesn't mean that we need to create new AMIs

File: aws/Full-PAS-Deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,1607 @@
...
+      LogGroupName: !Sub '${LogGroup}'
+      LogStreamName: CPMMachine/CPMRegistrationLog
+    DeletionPolicy: Retain
+  CPMSetLocalServiceLog:
+    Type: 'AWS::Logs::LogStream'
+    Properties:
+      LogGroupName: !Sub '${LogGroup}'
+      LogStreamName: CPMMachine/CPMSetLocalServiceLog
+    DeletionPolicy: Retain
+  PSMCfnInitLogStreamLog:
```
[the remark this answers is not in the document]
Comment (yogevh) [reply aws/Full-PAS-Deployment.yaml]: if it doesn't mean that we need to create new AMIsif it doesn't mean that we need to create new AMIs

File: aws/Full-PAS-Deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,1607 @@
...
+              command: !Sub >-
+                /opt/aws/bin/cfn-signal -e $? --stack ${AWS::StackId} --resource
+                PSMPMachine --region ${AWS::Region}
+    DependsOn:
+      - PSMMachine
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT10M
+    DeletionPolicy: Retain
+  InstancesSSMPolicy:
```
Comment (yogevh) [inline aws/Full-PAS-Deployment.yaml]: @avishayil why do we have this Policy and also AmazonEC2RoleforSSM on ComponentsInstanceProfile???
we need to examine this issue

File: aws/Full-PAS-Deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,1384 @@
...
+      SecurityGroupIds: !Ref PVWAInstanceSecurityGroups
+      SubnetId: !Ref PVWAInstanceSubnetId
+      ImageId: !FindInMap 
+        - RegionMap
+        - !Ref 'AWS::Region'
+        - PVWA
+      InstanceType: !Ref PVWAInstanceType
+      UserData: !Base64 
+        'Fn::Sub': >-
+          <script> cfn-init.exe -v -s ${AWS::StackId} -r PVWAMachine
```
Comment (avishayil) [inline aws/Full-PAS-Deployment.yaml]: @mliora you forgot this init :>

**Pull request merged**